Library/Homebrew/utils/curl.rb

```diff
@@ -140,9 +146,23 @@ def curl_args(
       (args + extra_args).map(&:to_s)
     end
 
+    sig {
+      params(
+        args:              T.untyped,
```

Comment:
I think these could be `String`, maybe `T.any(Pathname, String)` (or `T.any(Pathname, String, URI::Generic)`, etc.), depending how much work you want to do to update call sites.

Reply:
Yeh, we should definitely avoid `T.untyped` here.